python/tests/e2e/conftest.py

```diff
@@ -68,6 +77,66 @@ class VCRConfig(TypedDict):
     Useful for removing sensitive data from request bodies.
     """
 
+    before_record_request: Any
+    """Callback to sanitize requests before saving to cassette.
+
+    This function is called AFTER the real HTTP request is sent (with valid auth),
+    but BEFORE it's written to the cassette file. Use this to sanitize sensitive
+    headers without affecting the actual HTTP requests.
+    """
+
+    decode_compressed_response: bool
+    """Whether to decode compressed responses.
+
+    When False, VCR will preserve the exact bytes of compressed responses
+    without decoding them. This is important for maintaining data integrity
+    and avoiding issues with binary data or signatures.
+    """
+
+
+def sanitize_request(request: Any) -> Any:  # noqa: ANN401
+    """Sanitize sensitive headers in VCR request before recording to cassette.
+
+    This hook is called AFTER the real HTTP request is sent (with valid auth),
+    but BEFORE it's written to the cassette file. We deep copy the request
+    and replace sensitive headers with placeholders.
+
+    Args:
+        request: VCR request object to sanitize (Any type since VCR doesn't
+            provide typed request objects)
+
+    Returns:
+        Sanitized copy of the request safe for cassette storage
+    """
+    request = deepcopy(request)
+
+    common_sensitive_headers = [
+        "authorization",
+        "x-api-key",
+        "x-goog-api-key",
+        "anthropic-organization-id",
+    ]
+
+    for header in common_sensitive_headers:
+        if header in request.headers:
+            request.headers[header] = ["<filtered>"]
+
+    # Detect Bedrock requests and sanitize AWS SigV4 headers
+    if "bedrock-runtime" in request.uri:
+        aws_sigv4_headers = [
```

Comment:
Can we just include these in the headers to discard unconditionally? Presumably if the header is absent then it's no problem? 

Btw something I've considered is maybe we'd like to just drop all of the headers unconditionally regardless?